api/proto/teleport/legacy/types/types.proto

```diff
@@ -2447,6 +2447,13 @@ message RoleOptions {
     (gogoproto.jsontag) = "create_db_user",
     (gogoproto.customtype) = "BoolOption"
   ];
+
+  // CreateHostUserMode allows users to be automatically created on a
+  // host when not set to off
+  string CreateHostUserMode = 28 [
+    (gogoproto.jsontag) = "create_host_user_mode,omitempty",
+    (gogoproto.casttype) = "github.com/gravitational/teleport/api/constants.HostUserMode"
```

Comment:
Can we use a protobuf enum instead here? Keep in mind we want to move away from gogoproto _eventually_, so the less we depend on it the better.